install_influxdb.sh

```diff
@@ -90,30 +236,618 @@ URL="https://dl.influxdata.com/influxdb/releases/influxdb3-${EDITION_TAG}-${INFL
+            printf "%s\n\n" "$PULL_OUTPUT"
+
+            # Provide specific troubleshooting based on error content
+            printf "${BOLD}Possible causes:${NC}\n"
+            if echo "$PULL_OUTPUT" | grep -qi "denied\|unauthorized"; then
+                printf "  • Docker Hub authentication required or rate limited\n"
+                printf "  • Try: docker login\n"
+            elif echo "$PULL_OUTPUT" | grep -qi "network\|timeout\|connection"; then
+                printf "  • Network connectivity issue\n"
+                printf "  • Check your internet connection\n"
+            elif echo "$PULL_OUTPUT" | grep -qi "daemon"; then
+                printf "  • Docker daemon may not be responding\n"
+                printf "  • Try: docker info\n"
+            elif echo "$PULL_OUTPUT" | grep -qi "yaml\|control characters"; then
+                printf "  • Invalid YAML syntax in docker-compose.yml\n"
+                printf "  • Inspect file: cat %s/docker-compose.yml\n" "$DOCKER_DIR"
+            else
+                printf "  • Check Docker daemon status: docker info\n"
+                printf "  • Check internet connection\n"
+                printf "  • Check Docker Hub access\n"
             fi
-            if ! "$lsof_exec" -i:"$PORT" -t >/dev/null 2>&1; then
-                if [ "$show_progress" = "true" ]; then
-                    printf "└─${DIM} Found an available port: %s${NC}\n" "$PORT"
-                fi
-                break
+            printf "\n"
+            return 1
+        else
+            printf "${YELLOW}FAILED${NC}\n"
+            printf "│  ${YELLOW}Warning: %s will not be available${NC}\n" "$display_name"
+            printf "│  You can run this script again later to install it\n"
+            return 0
+        fi
+    fi
+}
+
+# --- Output Formatting ---
+
+# Utility function to filter Docker Compose output
+filter_docker_output() {
+    eval "$DOCKER_OUTPUT_FILTER" || true
+}
+
+# --- Filesystem Setup ---
+
+# Utility function to create docker directories with permissions
+create_docker_directories() {
+    DOCKER_DIR="$1"
+
+    # Create shared data directories (used by both Docker and binary)
+    mkdir -p "$HOME/.influxdb/data"
+    mkdir -p "$HOME/.influxdb/plugins"
+
+    # Create Docker-specific directories
+    mkdir -p "$DOCKER_DIR/explorer/db"
+    mkdir -p "$DOCKER_DIR/explorer/config"
+
+    chmod 700 "$DOCKER_DIR/explorer/db" 2>/dev/null || true
+    chmod 755 "$DOCKER_DIR/explorer/config" 2>/dev/null || true
+}
+
+# --- Health Checks ---
+
+# Utility function to wait for container to be ready
+wait_for_container_ready() {
+    CONTAINER_NAME="$1"
+    READY_MESSAGE="$2"
+    TIMEOUT="${3:-60}"
+    EDITION_TYPE="${4:-}"
+    LICENSE_TYPE="${5:-}"
+
+    printf "├─ Starting InfluxDB"
+    ELAPSED=0
+    EMAIL_MESSAGE_SHOWN=false
+
+    while [ $ELAPSED -lt $TIMEOUT ]; do
+        if docker logs "$CONTAINER_NAME" 2>&1 | grep -q "$READY_MESSAGE"; then
```

Comment:
I don't know if `docker` is translated, but if it is, this be an issue for users who don't use English on their computer. You should be able to handle this by setting `LC_ALL=C` at the top of the script.